### DISCUSSION TITLE
SATA Controller

### DISCUSSION THREAD

**Discussion by asefd** [mebs/CM4-NAS#23] [General]
Hi @mebs ,
Very great project. I like the design you made and doing all of this from scratch.
Which SATA Controller are you using now and do you have any problems with it?
Thanks

**Comment by mebs** [comment]
Hey there,

I used this exact card: https://www.aliexpress.com/item/32920676430.html

This specific listing is not available anymore, but the controller chip is what matters, it's a Marvell 88SE9215.

I have had absolutely no problem with it, it's been running for months now with 3 large HDDs as my only NAS and media server without any issues. I'm very happy with it!
Other controllers would probably work, check [Jeff Geerling's repo](https://github.com/geerlingguy/raspberry-pi-pcie-devices/issues) for a list of the ones he tested.

Now that the RPi OS kernel supports SATA controllers out of the box, you don't even have to recompile the kernel which is awesome.
